src/main/java/dave/Ui.java

```diff
@@ -5,62 +5,115 @@
     static final String NAME = "Dave";
+    /** Horizontal line divider for output. */
     static final String HORIZONTAL_LINE = "____________________________________________________________\n";
 
+    /** 
+     * Creates new UI object.
+     */
     public Ui() {
         this.sc = new Scanner(System.in);
     }
 
+    /**
+     * Shows the horizontal line divider in output.
+     */
     public void showHorizontalLine() {
         System.out.println(HORIZONTAL_LINE);
     }
 
+    /**
+     * Welcomes the user with a message.
+     */
     public void greet() {
         System.out.println(String.format("%s\n%s: Nice to meet you, I'm the ever-helpful %s!\nHow may I be of service today?\n%s",
         HORIZONTAL_LINE, NAME, NAME, HORIZONTAL_LINE));
         return;
     }
 
+    /**
+     * Says bye to the user with a message.
+     */
     public void sayBye() {
         System.out.println(String.format("%s\nDave: Thank you for your patronage, goodbye and have a nice day!\n%s",
         HORIZONTAL_LINE, HORIZONTAL_LINE));
         return;
     }
 
+    /**
+     * Shows that tasks from an existing task list have been loaded in the chatbot program, if any.
+     */
     public void showLoadingSuccess(int noOfTasks) {
         System.out.println(String.format("Dave has found %s existing tasks and loaded them.\n%s", noOfTasks, HORIZONTAL_LINE));
     }
 
+    /**
+     * Shows that there is no existing task list to load tasks from.
+     */
     public void showLoadingError() {
         System.out.println(String.format("Dave did not find any existing tasks to load.\n%s", HORIZONTAL_LINE));
     }
 
-    public void showLoadingError(String msg) {
+    /**
+     * Shows error from any user-chatbot interaction.
+     * 
+     * @param msg The error output.
+     */
+    public void showError(String msg) {
         System.out.println(msg);
     }
 
+    /**
+     * Reads user input.
+     * 
+     * @return The user input to the chatbot program.
+     */
     public String readCommand() {
         return this.sc.nextLine();
     }
 
+    /**
+     * Shows user feedback when a new task is successfully added into the task list.
+     * 
+     * @param newTask Information about the new task added.
+     * @param taskList Current task list.
+     */
     public void showTaskAdded(Task newTask, TaskList taskList) {
         System.out.println(String.format("%s\nDave added the task:\n  %s\nYou now have %d task(s).\n%s",
         HORIZONTAL_LINE, newTask.toString(), taskList.getNumberOfTasks(), HORIZONTAL_LINE));
     }
 
+    /**
+     * Shows user feedback when a task is successfully deleted from the task list.
+     * 
+     * @param deletedTask Information about the deleted task.
+     * @param taskList Current task list.
+     */
     public void showTaskDeleted(Task deletedTask, TaskList taskList) {
         System.out.println(String.format("%s\nDave has removed the task:\n  %s", HORIZONTAL_LINE, deletedTask.toString()));
```

Comment:
This line looks longer than the soft limit, you can consider splitting into two lines